app/src/main/java/org/mifos/mobile/ui/recent_transactions/RecentTransactionScreen.kt

```diff
@@ -0,0 +1,275 @@
+                val isNearBottom = lastVisibleItemIndex >= uiState.transactions.size - 5
+
+                if (isNearBottom) {
+                    viewModel.loadRecentTransactions(true, uiState.transactions.size)
+                }
+        }
+    }
+
+}
+
+
+@Composable
+fun RecentTransactionListItem(transaction: Transaction?) {
+    Row(modifier = Modifier.padding(8.dp), verticalAlignment = Alignment.CenterVertically) {
+        Image(
+            painter = painterResource(id = R.drawable.ic_local_atm_black_24dp),
+            contentDescription = stringResource(id = R.string.atm_icon),
+            modifier = Modifier.size(39.dp)
+        )
+
+        Spacer(modifier = Modifier.width(8.dp))
+
+        Column(modifier = Modifier.weight(1f)) {
+            Text(
+                text = Utils.formatTransactionType(transaction?.type?.value),
+                style = MaterialTheme.typography.bodyMedium,
+                color = MaterialTheme.colorScheme.onSurface
+            )
+
+            Row {
+                Text(
+                    text = stringResource(
+                        id = R.string.string_and_string,
+                        transaction?.currency?.displaySymbol ?: transaction?.currency?.code ?: "",
+                        CurrencyUtil.formatCurrency(
+                            MifosSelfServiceApp.context,
+                            transaction?.amount ?: 0.0,
+                        )
+                    ),
+                    style = MaterialTheme.typography.labelMedium,
+                    modifier = Modifier
+                        .weight(1f)
+                        .alpha(0.7f),
+                    color = MaterialTheme.colorScheme.onSurface
+                )
+                Text(
+                    text = DateHelper.getDateAsString(transaction?.submittedOnDate),
+                    style = MaterialTheme.typography.labelMedium,
+                    modifier = Modifier.alpha(0.7f),
+                    color = MaterialTheme.colorScheme.onSurface
+                )
+            }
+
+        }
+    }
+}
+
+
+class RecentTransactionScreenPreviewProvider : PreviewParameterProvider<RecentTransactionUiState> {
+
+    var tList : ArrayList<Transaction> = ArrayList<Transaction>()
+    override val values: Sequence<RecentTransactionUiState>
+        get() = sequenceOf(
+            RecentTransactionUiState.RecentTransactions(tList),
+            RecentTransactionUiState.Loading,
+            RecentTransactionUiState.LoadMoreRecentTransactions(tList),
+            RecentTransactionUiState.Error(R.string.recent_transactions),
+            RecentTransactionUiState.EmptyTransaction,
+            RecentTransactionUiState.Initial
+        )
+}
+@Preview(showSystemUi = true, showBackground = true)
+@Composable
+private fun RecentTransactionScreenPreview(
+    @PreviewParameter(RecentTransactionScreenPreviewProvider::class) recentTransactionUiState: RecentTransactionUiState
+) {
+/*
+    MifosMobileTheme {
```

Comment:
Commented code ?

Reply:
that is for the preview, i should pass a view model io it, is there any way i can pass a dummy view model into it ??

Reply:
No Akash passing a viewmodel to preview is a very bad Idea.Observe how we separate concerns in our prs it will give you idea.